### PR TITLE
Add vector forward mode in docs

### PR DESCRIPTION
**Pull request by vaithak**
Moved the vector mode doc from the compiler-research webpage. (also, condensed it a lot).

### CODE REVIEW

File: docs/userDocs/source/user/CoreConcepts.rst [hunk 1]
```diff
@@ -127,6 +127,52 @@ Substituting `s = z` we will get `sz` = 1
 Thus we don't need to run the program twice for each input. However, as mentioned
 above the only drawback is we need to re-run the program for a different output.
 
+
+Vectorized Forward Mode Automatic Differentiation
```
Comment (vgvassilev) [inline docs/userDocs/source/user/CoreConcepts.rst:131]: Could we give examples how to use that within clad with the proper api, etc. Maybe that usage documentation needs to go in a different section.

Comment (vaithak) [reply docs/userDocs/source/user/CoreConcepts.rst:131]: We already have a usage section here: https://github.com/vgvassilev/clad/blob/master/docs/userDocs/source/user/UsingVectorMode.rst.

Comment (vgvassilev) [reply docs/userDocs/source/user/CoreConcepts.rst:131]: Okay, let's move forward then!

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**